packages/grid/src/GridRenderer.ts

```diff
@@ -3009,6 +3050,154 @@ export class GridRenderer {
 
     context.translate(-barLeft, -barTop);
   }
+
+  /**
+   * Gets the token boxes that are visible in the cell
+   * @param column The visible column
+   * @param row The visible row
+   * @param state The GridRenderState
+   * @returns An array of TokenBox of visible tokens or empty array
+   */
+  getTokenBoxesForVisibleCell(
+    column: VisibleIndex,
+    row: VisibleIndex,
+    state: GridRenderState
+  ): TokenBox[] {
+    const { metrics, context, model, theme } = state;
+    const { modelRows, modelColumns } = metrics;
+
+    if (context == null || metrics == null) {
+      return [];
+    }
+
+    const modelRow = getOrThrow(modelRows, row);
+    const modelColumn = getOrThrow(modelColumns, column);
+
+    const text = model.textForCell(modelColumn, modelRow);
+    const { width: textWidth, x: textX, y: textY } = this.getTextRenderMetrics(
+      state,
+      column,
+      row
+    );
+
+    const { fontWidths } = metrics;
+
+    // Set the font and baseline and change it back after
+    context.save();
+    context.font = theme.font;
+    context.textBaseline = 'middle';
```

Comment:
Instead of setting font/textBaseline again, call `this.configureContext(context, state)` to make sure the context is configured correctly.

Though I went down a bit of a rabbit hole... I was wondering why you needed to set the `textBaseline` considering we never set it to anything else. Digging into it a bit, it looks like the baseline gets reset when we update the width/height of the canvas: https://stackoverflow.com/questions/49094139/canvas-context-gets-reset-when-canvas-width-height-are-changed

We do that in [Grid.updateCanvasScale](https://github.com/deephaven/web-client-ui/blob/46c5e2461ab2344bd62d482ddcfd179c69e81ce0/packages/grid/src/Grid.tsx#L819).

We should really be configuring the context there right after we scale it to ensure it's always got the properties we expect. But `configureContext` takes the `GridRenderState`, which we don't have yet in `updateCanvasScale`... So I think a more complete fix:
- In `Grid.updateCanvas`, add a call to `this.updateRenderState()` after setting the context scale, and then call `this.renderer.configureContext(context, renderState)`.
- In `drawCanvas`, don't call `updateRenderState()` anymore, just use `this.renderState` (since it has already been updated)
- Mark `updateCanvasScale` and `drawCanvas` as `private` methods - we should only really call `requestUpdateCanvas()` or `updateCanvas()` externally.
  - There's a couple places where `updateCanvasScale` is called before calling `updateCanvas()`, those calls can simply be removed from: `componentDidMount()`, `handleResize()`, and `IrisGrid.handleAnimationLoop()`